fdbrpc/genericactors.actor.h

```diff
@@ -70,6 +72,100 @@ Future<REPLY_TYPE(Req)> retryBrokenPromise(RequestStream<Req> to, Req request, T
 	}
 }
 
+template <class Req>
+Future<ErrorOr<REPLY_TYPE(Req)>> tryGetReplyFromHostname(RequestStream<Req>* to,
+                                                         Req request,
+                                                         Hostname hostname,
+                                                         WellKnownEndpoints token) {
+	// A wrapper of tryGetReply(request), except that the request is sent to an address resolved from a hostname.
+	// If resolving fails, return lookup_failed().
+	// Otherwise, return tryGetReply(request).
+	try {
+		hostname.resolveBlocking();
+	} catch (...) {
+		return ErrorOr<REPLY_TYPE(Req)>(lookup_failed());
+	}
+	Optional<NetworkAddress> address = hostname.resolvedAddress;
+	*to = RequestStream<Req>(Endpoint::wellKnown({ address.get() }, token));
+	return to->tryGetReply(request);
+}
+
+template <class Req>
+Future<ErrorOr<REPLY_TYPE(Req)>> tryGetReplyFromHostname(RequestStream<Req>* to,
+                                                         Req request,
+                                                         Hostname hostname,
+                                                         WellKnownEndpoints token,
+                                                         TaskPriority taskID) {
+	// A wrapper of tryGetReply(request), except that the request is sent to an address resolved from a hostname.
+	// If resolving fails, return lookup_failed().
+	// Otherwise, return tryGetReply(request).
+	try {
+		hostname.resolveBlocking();
```

Comment:
I think you want to use `wait(hostname.resolve())` here. Otherwise, this function is all synchronous.